File: app/src/main/java/com/qltech/bws/DashboardModule/Playlist/ViewAllPlaylistFragment.java
package com.qltech.bws.DashboardModule.Playlist;

import android.content.Context;
import android.content.SharedPreferences;
import android.os.AsyncTask;
import android.os.Bundle;
import android.view.KeyEvent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.WindowManager;
import android.widget.ImageView;
import android.widget.LinearLayout;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.fragment.app.FragmentManager;
import androidx.recyclerview.widget.DefaultItemAnimator;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.engine.DiskCacheStrategy;
import com.google.gson.Gson;
import com.qltech.bws.BWSApplication;
import com.qltech.bws.DashboardModule.Models.ViewAllAudioListModel;
import com.qltech.bws.DashboardModule.Models.ViewAllPlayListModel;
import com.qltech.bws.DashboardModule.TransparentPlayer.Fragments.TransparentPlayerFragment;
import com.qltech.bws.R;
import com.qltech.bws.RoomDataBase.DatabaseClient;
import com.qltech.bws.RoomDataBase.DownloadAudioDetails;
import com.qltech.bws.RoomDataBase.DownloadPlaylistDetails;
import com.qltech.bws.Utility.APIClient;
import com.qltech.bws.Utility.CONSTANTS;
import com.qltech.bws.Utility.MeasureRatio;
import com.qltech.bws.databinding.FragmentViewAllPlaylistBinding;
import com.qltech.bws.databinding.PlaylistCustomLayoutBinding;

import java.util.ArrayList;
import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

import static android.content.Context.MODE_PRIVATE;
import static com.qltech.bws.DashboardModule.Activities.DashboardActivity.player;
import static com.qltech.bws.DashboardModule.Search.SearchFragment.comefrom_search;
import static com.qltech.bws.DashboardModule.Audio.AudioFragment.IsLock;
import static com.qltech.bws.Utility.MusicService.isMediaStart;
import static com.qltech.bws.Utility.MusicService.isPause;
import static com.qltech.bws.Utility.MusicService.isPrepare;
import static com.qltech.bws.Utility.MusicService.stopMedia;

public class ViewAllPlaylistFragment extends Fragment {
    FragmentViewAllPlaylistBinding binding;
    String GetLibraryID, Name, UserID, AudioFlag, MyDownloads;
    public static String GetPlaylistLibraryID = "";
    List<DownloadPlaylistDetails> playlistList;
    List<DownloadAudioDetails> playlistWiseAudioDetails = new ArrayList<>();
    View view;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = DataBindingUtil.inflate(inflater, R.layout.fragment_view_all_playlist, container, false);
        view = binding.getRoot();
        Glide.with(getActivity()).load(R.drawable.loading).asGif().into(binding.ImgV);
        SharedPreferences shared1 = getActivity().getSharedPreferences(CONSTANTS.PREF_KEY_LOGIN, Context.MODE_PRIVATE);
        UserID = (shared1.getString(CONSTANTS.PREF_KEY_UserID, ""));
        SharedPreferences shared = getActivity().getSharedPreferences(CONSTANTS.PREF_KEY_AUDIO, Context.MODE_PRIVATE);
        AudioFlag = shared.getString(CONSTANTS.PREF_KEY_AudioFlag, "0");
        if (getArguments() != null) {
            GetLibraryID = getArguments().getString("GetLibraryID");
            Name = getArguments().getString("Name");
            MyDownloads = getArguments().getString("MyDownloads");
        }

        view.setFocusableInTouchMode(true);
        view.requestFocus();
        view.setOnKeyListener((v, keyCode, event) -> {
            if (keyCode == KeyEvent.KEYCODE_BACK) {
                callBack();
                return true;
            }
            return false;
        });

        binding.llBack.setOnClickListener(view1 -> {
            callBack();
        });

        GridLayoutManager manager = new GridLayoutManager(getActivity(), 2);
        binding.rvMainAudio.setItemAnimator(new DefaultItemAnimator());
        binding.rvMainAudio.setLayoutManager(manager);
        if (MyDownloads.equalsIgnoreCase("1")) {
            playlistList = new ArrayList<>();
            GetAllMedia();
        } else {
            prepareData();
        }
        return view;
    }

    private void GetAllMedia() {
        class GetTask extends AsyncTask<Void, Void, Void> {
            @Override
            protected Void doInBackground(Void... voids) {
                playlistList = DatabaseClient
                        .getInstance(getActivity())
                        .getaudioDatabase()
                        .taskDao()
                        .getAllPlaylist();
                return null;
            }

            @Override
            protected void onPostExecute(Void aVoid) {
                binding.tvTitle.setText("My Downloads");

                ArrayList<ViewAllPlayListModel.ResponseData.Detail> listModelList = new ArrayList<>();
                for (int i = 0; i < playlistList.size(); i++) {
                    ViewAllPlayListModel.ResponseData.Detail detail = new ViewAllPlayListModel.ResponseData.Detail();

                    detail.setPlaylistID(playlistList.get(i).getPlaylistID());
                    detail.setPlaylistDesc(playlistList.get(i).getPlaylistDesc());
                    detail.setPlaylistName(playlistList.get(i).getPlaylistName());
                    detail.setPlaylistImage(playlistList.get(i).getPlaylistImage());
                    listModelList.add(detail);
                }
                PlaylistAdapter adapter = new PlaylistAdapter(listModelList, IsLock);
                binding.rvMainAudio.setAdapter(adapter);
                super.onPostExecute(aVoid);
            }
        }
        GetTask getTask = new GetTask();
        getTask.execute();
    }

    private void callBack() {
        Fragment audioFragment = new PlaylistFragment();
        FragmentManager fragmentManager1 = getActivity().getSupportFragmentManager();
        fragmentManager1.beginTransaction()
                .replace(R.id.flContainer, audioFragment)
                .commit();
        Bundle bundle = new Bundle();
        audioFragment.setArguments(bundle);
    }

    @Override
    public void onResume() {
        super.onResume();
//        prepareData();
    }

    private void prepareData() {
        try {
            if (IsLock.equalsIgnoreCase("1") && !AudioFlag.equalsIgnoreCase("AppointmentDetailList")) {
                SharedPreferences sharedm = getActivity().getSharedPreferences(CONSTANTS.PREF_KEY_AUDIO, Context.MODE_PRIVATE);
                SharedPreferences.Editor editorr = sharedm.edit();
                editorr.remove(CONSTANTS.PREF_KEY_modelList);
                editorr.remove(CONSTANTS.PREF_KEY_position);
                editorr.remove(CONSTANTS.PREF_KEY_queuePlay);
                editorr.remove(CONSTANTS.PREF_KEY_audioPlay);
                editorr.remove(CONSTANTS.PREF_KEY_AudioFlag);
                editorr.remove(CONSTANTS.PREF_KEY_PlaylistId);
                editorr.remove(CONSTANTS.PREF_KEY_myPlaylist);
                editorr.clear();
                editorr.commit();
            }
            SharedPreferences shared = getActivity().getSharedPreferences(CONSTANTS.PREF_KEY_AUDIO, MODE_PRIVATE);
            AudioFlag = shared.getString(CONSTANTS.PREF_KEY_AudioFlag, "0");
            if (!AudioFlag.equalsIgnoreCase("0")) {
                Fragment fragment = new TransparentPlayerFragment();
                FragmentManager fragmentManager1 = getActivity().getSupportFragmentManager();
                fragmentManager1.beginTransaction()
                        .add(R.id.flContainer, fragment)
                        .commit();
                LinearLayout.LayoutParams params = new LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT);
                params.setMargins(4, 6, 4, 260);
                binding.llSpace.setLayoutParams(params);
            } else {
                LinearLayout.LayoutParams params = new LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT);
                params.setMargins(4, 6, 4, 50);
                binding.llSpace.setLayoutParams(params);
            }
        } catch (Exception e) {
            e.printStackTrace();
        }

        if (BWSApplication.isNetworkConnected(getActivity())) {
            try {
                showProgressBar();
                Call<ViewAllPlayListModel> listCall = APIClient.getClient().getViewAllPlayLists(UserID, GetLibraryID);
                listCall.enqueue(new Callback<ViewAllPlayListModel>() {
                    @Override
                    public void onResponse(Call<ViewAllPlayListModel> call, Response<ViewAllPlayListModel> response) {
                        if (response.isSuccessful()) {
                            hideProgressBar();
                            ViewAllPlayListModel listModel = response.body();
                            binding.tvTitle.setText(listModel.getResponseData().getView());
                            PlaylistAdapter adapter = new PlaylistAdapter(listModel.getResponseData().getDetails(), listModel.getResponseData().getIsLock());
                            binding.rvMainAudio.setAdapter(adapter);
                        }
                    }

                    @Override
                    public void onFailure(Call<ViewAllPlayListModel> call, Throwable t) {
                        hideProgressBar();
                    }
                });
            } catch (Exception e) {
                e.printStackTrace();
            }

        } else {
            BWSApplication.showToast(getString(R.string.no_server_found), getActivity());
        }
    }

    private void hideProgressBar() {
        try {
            binding.progressBarHolder.setVisibility(View.GONE);
            binding.ImgV.setVisibility(View.GONE);
            getActivity().getWindow().clearFlags(WindowManager.LayoutParams.FLAG_NOT_TOUCHABLE);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    private void showProgressBar() {
        try {
            binding.progressBarHolder.setVisibility(View.VISIBLE);
            getActivity().getWindow().setFlags(WindowManager.LayoutParams.FLAG_NOT_TOUCHABLE, WindowManager.LayoutParams.FLAG_NOT_TOUCHABLE);
            binding.ImgV.setVisibility(View.VISIBLE);
            binding.ImgV.invalidate();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    public class PlaylistAdapter extends RecyclerView.Adapter<PlaylistAdapter.MyViewHolder> {
        private List<ViewAllPlayListModel.ResponseData.Detail> listModelList;
        String IsLock;

        public PlaylistAdapter(List<ViewAllPlayListModel.ResponseData.Detail> listModelList, String IsLock) {
            this.listModelList = listModelList;
            this.IsLock = IsLock;
        }

        @NonNull
        @Override
        public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
            PlaylistCustomLayoutBinding v = DataBindingUtil.inflate(LayoutInflater.from(parent.getContext())
                    , R.layout.playlist_custom_layout, parent, false);
            return new MyViewHolder(v);
        }

        @Override
        public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {
            MeasureRatio measureRatio = BWSApplication.measureRatio(getActivity(), 0,
                    1, 1, 0.46f, 0);
            holder.binding.ivRestaurantImage.getLayoutParams().height = (int) (measureRatio.getHeight() * measureRatio.getRatio());
            holder.binding.ivRestaurantImage.getLayoutParams().width = (int) (measureRatio.getWidthImg() * measureRatio.getRatio());
            holder.binding.ivRestaurantImage.setScaleType(ImageView.ScaleType.FIT_XY);

            holder.binding.tvPlaylistName.setText(listModelList.get(position).getPlaylistName());
            Glide.with(getActivity()).load(listModelList.get(position).getPlaylistImage()).thumbnail(0.05f)
                    .diskCacheStrategy(DiskCacheStrategy.ALL).skipMemoryCache(false).into(holder.binding.ivRestaurantImage);

            if (IsLock.equalsIgnoreCase("1")) {
                holder.binding.ivLock.setVisibility(View.VISIBLE);
            } else if (IsLock.equalsIgnoreCase("0") || IsLock.equalsIgnoreCase("")) {
                holder.binding.ivLock.setVisibility(View.GONE);
            }

            holder.binding.rlMainLayout.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    if (IsLock.equalsIgnoreCase("1")) {
                        holder.binding.ivLock.setVisibility(View.VISIBLE);
                        BWSApplication.showToast("Please re-activate your membership plan", getActivity());
                    } else if (IsLock.equalsIgnoreCase("0") || IsLock.equalsIgnoreCase("")) {
                        holder.binding.ivLock.setVisibility(View.GONE);
                        if (MyDownloads.equalsIgnoreCase("1")) {
                            getMedia(listModelList.get(position).getPlaylistID());

                        } else {
                            Bundle bundle = new Bundle();
                            comefrom_search = 2;
                            GetPlaylistLibraryID = GetLibraryID;
                            Fragment myPlaylistsFragment = new MyPlaylistsFragment();
                            FragmentManager fragmentManager1 = getActivity().getSupportFragmentManager();
                            bundle.putString("New", "0");
                            bundle.putString("PlaylistID", listModelList.get(position).getPlaylistID());
                            bundle.putString("PlaylistName", listModelList.get(position).getPlaylistName());
                            bundle.putString("PlaylistImage", listModelList.get(position).getPlaylistImage());
                            bundle.putString("MyDownloads", MyDownloads);
                            myPlaylistsFragment.setArguments(bundle);
                            fragmentManager1.beginTransaction()
                                    .replace(R.id.flContainer, myPlaylistsFragment)
                                    .commit();
                        }
                    }
                }
            });
        }

        @Override
        public int getItemCount() {
            return listModelList.size();
        }

        public class MyViewHolder extends RecyclerView.ViewHolder {
            PlaylistCustomLayoutBinding binding;

            public MyViewHolder(PlaylistCustomLayoutBinding binding) {
                super(binding.getRoot());
                this.binding = binding;
            }
        }
    }

    private void getMedia(String playlistID) {
        class GetMedia extends AsyncTask<Void, Void, Void> {
            @Override
            protected Void doInBackground(Void... voids) {
                playlistWiseAudioDetails = DatabaseClient
                        .getInstance(getActivity())
                        .getaudioDatabase()
                        .taskDao()
                        .getAllAudioByPlaylist(playlistID);
                return null;
            }

            @Override
            protected void onPostExecute(Void aVoid) { player = 1;
                if (isPrepare || isMediaStart || isPause) {
                    stopMedia();
                }
                isPause = false;
                isMediaStart = false;
                isPrepare = false;
                SharedPreferences shared = getActivity().getSharedPreferences(CONSTANTS.PREF_KEY_AUDIO, Context.MODE_PRIVATE);
                SharedPreferences.Editor editor = shared.edit();
                Gson gson = new Gson();
                String json = gson.toJson(playlistWiseAudioDetails);
                editor.putString(CONSTANTS.PREF_KEY_modelList, json);
                editor.putInt(CONSTANTS.PREF_KEY_position, 0);
                editor.putBoolean(CONSTANTS.PREF_KEY_queuePlay, false);
                editor.putBoolean(CONSTANTS.PREF_KEY_audioPlay, true);
                editor.putString(CONSTANTS.PREF_KEY_PlaylistId, playlistID);
                editor.putString(CONSTANTS.PREF_KEY_myPlaylist, "");
                editor.putString(CONSTANTS.PREF_KEY_AudioFlag, "SubPlayList");
                editor.commit();
                try {
                    Fragment fragment = new TransparentPlayerFragment();
                    FragmentManager fragmentManager1 = getActivity().getSupportFragmentManager();
                    fragmentManager1.beginTransaction()
                            .add(R.id.flContainer, fragment)
                            .commit();
                } catch (Exception e) {
                    e.printStackTrace();
                }

                super.onPostExecute(aVoid);
            }
        }
        GetMedia st = new GetMedia();
        st.execute();
    }
}

File: app/src/main/java/com/brainwellnessspa/dashboardModule/manage/ViewAllPlaylistFragment.kt
package com.brainwellnessspa.dashboardModule.manage

import android.annotation.SuppressLint
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.KeyEvent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.DefaultItemAnimator
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.brainwellnessspa.BWSApplication
import com.brainwellnessspa.dashboardModule.models.SegmentPlaylist
import com.brainwellnessspa.dashboardModule.models.ViewAllPlayListModel
import com.brainwellnessspa.downloadModule.activities.DownloadPlaylistActivity
import com.brainwellnessspa.R
import com.brainwellnessspa.dashboardModule.activities.AddPlaylistActivity
import com.brainwellnessspa.databinding.FragmentViewAllPlaylistBinding
import com.brainwellnessspa.databinding.PlaylistViewAllLayoutBinding
import com.brainwellnessspa.roomDataBase.DownloadPlaylistDetailsUnique
import com.brainwellnessspa.utility.APINewClient
import com.brainwellnessspa.utility.CONSTANTS
import com.bumptech.glide.Glide
import com.bumptech.glide.Priority
import com.bumptech.glide.load.engine.DiskCacheStrategy
import com.bumptech.glide.load.resource.bitmap.RoundedCorners
import com.bumptech.glide.request.RequestOptions
import com.google.gson.Gson
import com.segment.analytics.Properties
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import java.util.*

class ViewAllPlaylistFragment : Fragment() {
    lateinit var binding: FragmentViewAllPlaylistBinding
    var getLibraryId: String? = null
    var name: String? = null
    var coUserId: String? = null
    var userId: String? = null
    var userName: String? = null
    var audioFlag: String? = null
    var myDownloads: String? = null
    var screenView: String? = ""
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
    ): View {
        binding =
            DataBindingUtil.inflate(inflater, R.layout.fragment_view_all_playlist, container, false)
        val shared1 = requireActivity()
            .getSharedPreferences(CONSTANTS.PREFE_ACCESS_SIGNIN_COUSER, Context.MODE_PRIVATE)
        userId = shared1.getString(CONSTANTS.PREFE_ACCESS_mainAccountID, "")
        coUserId = shared1.getString(CONSTANTS.PREFE_ACCESS_UserId, "")
        userName = shared1.getString(CONSTANTS.PREFE_ACCESS_NAME, "")
        val shared =
            requireActivity().getSharedPreferences(CONSTANTS.PREF_KEY_PLAYER, Context.MODE_PRIVATE)
        audioFlag = shared.getString(CONSTANTS.PREF_KEY_AudioPlayerFlag, "0")
        BWSApplication.DB = BWSApplication.getAudioDataBase(activity)
        if (arguments != null) {
            getLibraryId = requireArguments().getString("GetLibraryID")
            name = requireArguments().getString("Name")
            myDownloads = requireArguments().getString("MyDownloads")
        }
        requireView().isFocusableInTouchMode = true
        requireView().requestFocus()
        requireView().setOnKeyListener { _: View?, keyCode: Int, event: KeyEvent ->
            if (event.action == KeyEvent.ACTION_UP && keyCode == KeyEvent.KEYCODE_BACK) {
                callBack()
                return@setOnKeyListener true
            }
            false
        }
        binding.llBack.setOnClickListener { callBack() }
        val manager = GridLayoutManager(activity, 2)
        binding.rvMainAudio.itemAnimator = DefaultItemAnimator()
        binding.rvMainAudio.layoutManager = manager
        return binding.root
    }

    @SuppressLint("SetTextI18n")
    private fun getAllMedia() {
        BWSApplication.DB.taskDao()
            .getAllPlaylist1(coUserId)
            .observe(
                requireActivity(),
                { audioList: List<DownloadPlaylistDetailsUnique> ->
                    binding.tvTitle.text = "My Downloads"
                    screenView = "My Downloads"
                    val listModelList = ArrayList<ViewAllPlayListModel.ResponseData.Detail>()
                    for (i in audioList.indices) {
                        val detail = ViewAllPlayListModel.ResponseData.Detail()
                        detail.totalAudio = audioList[i].totalAudio
                        detail.totalhour = audioList[i].totalhour
                        detail.totalminute = audioList[i].totalminute
                        detail.playlistID = audioList[i].playlistID
                        detail.playlistDesc = audioList[i].playlistDesc
                        detail.playlistMastercat = audioList[i].playlistMastercat
                        detail.playlistSubcat = audioList[i].playlistSubcat
                        detail.playlistName = audioList[i].playlistName
                        detail.playlistImage = audioList[i].playlistImage
                        //                detail.setPlaylistImageDetails("");
                        detail.created = audioList[i].created
                        listModelList.add(detail)
                    }
                    val p = Properties()
                    val section = ArrayList<SegmentPlaylist>()
                    for (i in audioList.indices) {
                        val e =
                            SegmentPlaylist()
                        e.playlistId = audioList[i].playlistID
                        e.playlistName = audioList[i].playlistName
                        e.playlistType = audioList[i].created
                        e.playlistDuration = (audioList[i].totalhour
                                + "h "
                                + audioList[i].totalminute
                                + "m")
                        e.audioCount = audioList[i].totalAudio
                        section.add(e)
                    }
                    p.putValue("userId", coUserId)
                    val gson = Gson()
                    p.putValue("playlists", gson.toJson(section))
                    p.putValue("section", screenView)
                    BWSApplication.addToSegment(
                        "View All Playlist Screen Viewed",
                        p,
                        CONSTANTS.screen
                    )
                    val adapter = PlaylistAdapter(listModelList)
                    binding.rvMainAudio.adapter = adapter
                })
    }

    private fun callBack() {
        val audioFragment: Fragment = MainPlaylistFragment()
        val fragmentManager1 = requireActivity().supportFragmentManager
        fragmentManager1.beginTransaction().replace(R.id.flContainer, audioFragment).commit()
    }

    override fun onResume() {
        if (myDownloads.equals("1", ignoreCase = true)) {
            getAllMedia()
        } else {
            prepareData()
        }
        super.onResume()
    }

    private fun prepareData() {
        /*     try {
        SharedPreferences shared1 = getActivity().getSharedPreferences(CONSTANTS.PREF_KEY_PLAYER, Context.MODE_PRIVATE);
        audioFlag = shared1.getString(CONSTANTS.PREF_KEY_AudioPlayerFlag, "0");
        SharedPreferences shared2 = getActivity().getSharedPreferences(CONSTANTS.PREF_KEY_LOGIN, Context.MODE_PRIVATE);
        String UnlockAudioLists = shared2.getString(CONSTANTS.PREF_KEY_UnLockAudiList, "");
        Gson gson1 = new Gson();
        Type type1 = new TypeToken<List<String>>() {
        }.getType();
        List<String> UnlockAudioList = gson1.fromJson(UnlockAudioLists, type1);
        if (!IsLock.equalsIgnoreCase("0") && (audioFlag.equalsIgnoreCase("MainAudioList")
                || audioFlag.equalsIgnoreCase("ViewAllAudioList"))) {
            String audioID = "";
            SharedPreferences shared = getActivity().getSharedPreferences(CONSTANTS.PREF_KEY_PLAYER, Context.MODE_PRIVATE);
            Gson gson = new Gson();
            String json = shared.getString(CONSTANTS.PREF_KEY_PlayerAudioList, String.valueOf(gson));
            Type type = new TypeToken<ArrayList<MainPlayModel>>() {
            }.getType();
            ArrayList<MainPlayModel> arrayList = gson.fromJson(json, type);

            if (arrayList.get(0).getAudioFile().equalsIgnoreCase("")) {
                arrayList.remove(0);
            }
            audioID = arrayList.get(0).getID();

            if (UnlockAudioList.contains(audioID)) {
            } else {
                SharedPreferences sharedm = getActivity().getSharedPreferences(CONSTANTS.PREF_KEY_PLAYER, Context.MODE_PRIVATE);
                SharedPreferences.Editor editorr = sharedm.edit();
                editorr.remove(CONSTANTS.PREF_KEY_MainAudioList);
                editorr.remove(CONSTANTS.PREF_KEY_PlayerAudioList);
                editorr.remove(CONSTANTS.PREF_KEY_PlayerPosition);
                editorr.remove(CONSTANTS.PREF_KEY_queuePlay);
                editorr.remove(CONSTANTS.PREF_KEY_audioPlay);
                editorr.remove(CONSTANTS.PREF_KEY_AudioPlayerFlag);
                editorr.remove(CONSTANTS.PREF_KEY_PlaylistId);
                editorr.remove(CONSTANTS.PREF_KEY_myPlaylist);
                editorr.clear();
                editorr.commit();
                callNewPlayerRelease();
            }
        } else if (!IsLock.equalsIgnoreCase("0") && !audioFlag.equalsIgnoreCase("AppointmentDetailList")) {
            SharedPreferences sharedm = getActivity().getSharedPreferences(CONSTANTS.PREF_KEY_PLAYER, Context.MODE_PRIVATE);
            SharedPreferences.Editor editorr = sharedm.edit();
            editorr.remove(CONSTANTS.PREF_KEY_MainAudioList);
            editorr.remove(CONSTANTS.PREF_KEY_PlayerAudioList);
            editorr.remove(CONSTANTS.PREF_KEY_PlayerPosition);
            editorr.remove(CONSTANTS.PREF_KEY_queuePlay);
            editorr.remove(CONSTANTS.PREF_KEY_audioPlay);
            editorr.remove(CONSTANTS.PREF_KEY_AudioPlayerFlag);
            editorr.remove(CONSTANTS.PREF_KEY_PlaylistId);
            editorr.remove(CONSTANTS.PREF_KEY_myPlaylist);
            editorr.clear();
            editorr.commit();
            callNewPlayerRelease();

        }
        SharedPreferences shared = getActivity().getSharedPreferences(CONSTANTS.PREF_KEY_PLAYER, Context.MODE_PRIVATE);
        audioFlag = shared.getString(CONSTANTS.PREF_KEY_AudioPlayerFlag, "0");
        if (!audioFlag.equalsIgnoreCase("0")) {

            Fragment fragment = new MiniPlayerFragment();
            FragmentManager fragmentManager1 = getActivity().getSupportFragmentManager();
            fragmentManager1.beginTransaction()
                    .add(R.id.flContainer, fragment)
                    .commit();
            LinearLayout.LayoutParams params = new LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT);
            params.setMargins(4, 6, 4, 280);
            binding.llSpace.setLayoutParams(params);
        } else {
            LinearLayout.LayoutParams params = new LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT);
            params.setMargins(4, 6, 4, 50);
            binding.llSpace.setLayoutParams(params);
        }
    } catch (Exception e) {
        e.printStackTrace();
    }*/
        if (BWSApplication.isNetworkConnected(activity)) {
            try {
                BWSApplication.showProgressBar(
                    binding.progressBar, binding.progressBarHolder, activity
                )
                val listCall = APINewClient.getClient().getViewAllPlayLists(coUserId, getLibraryId)
                listCall.enqueue(
                    object : Callback<ViewAllPlayListModel?> {
                        override fun onResponse(
                            call: Call<ViewAllPlayListModel?>,
                            response: Response<ViewAllPlayListModel?>
                        ) {
                            try {
                                val listModel = response.body()
                                if (listModel != null) {
                                    if (listModel
                                            .responseCode
                                            .equals(
                                                getString(R.string.ResponseCodesuccess),
                                                ignoreCase = true
                                            )
                                    ) {
                                        BWSApplication.hideProgressBar(
                                            binding.progressBar,
                                            binding.progressBarHolder,
                                            activity
                                        )
                                        binding.tvTitle.text = listModel.responseData!!.view
                                        screenView = listModel.responseData!!.view
                                        val p = Properties()
                                        val section = ArrayList<SegmentPlaylist>()
                                        for (i in listModel.responseData!!.details!!.indices) {
                                            val e =
                                                SegmentPlaylist()
                                            e.playlistId =
                                                listModel.responseData!!.details!![i].playlistID
                                            e.playlistName =
                                                listModel.responseData!!.details!![i].playlistName
                                            e.playlistType =
                                                listModel.responseData!!.details!![i].created
                                            e.playlistDuration =
                                                (listModel.responseData!!.details!![i].totalhour
                                                        + "h "
                                                        + listModel.responseData!!.details!![i].totalminute
                                                        + "m")
                                            e.audioCount =
                                                listModel.responseData!!.details!![i].totalAudio
                                            section.add(e)
                                        }
                                        p.putValue("userId", coUserId)
                                        val gson = Gson()
                                        p.putValue("playlists", gson.toJson(section))
                                        p.putValue("section", screenView)
                                        BWSApplication.addToSegment(
                                            "View All Playlist Screen Viewed", p, CONSTANTS.screen
                                        )
                                        val adapter = PlaylistAdapter(
                                            listModel.responseData!!.details!!
                                        )
                                        binding.rvMainAudio.adapter = adapter
                                    }
                                }
                            } catch (e: Exception) {
                                e.printStackTrace()
                            }
                        }

                        override fun onFailure(
                            call: Call<ViewAllPlayListModel?>, t: Throwable
                        ) {
                            BWSApplication.hideProgressBar(
                                binding.progressBar, binding.progressBarHolder, activity
                            )
                        }
                    })
            } catch (e: Exception) {
                e.printStackTrace()
            }
        } else {
            BWSApplication.showToast(getString(R.string.no_server_found), activity)
        }
    }

    inner class PlaylistAdapter(private val listModelList: List<ViewAllPlayListModel.ResponseData.Detail>) :
        RecyclerView.Adapter<PlaylistAdapter.MyViewHolder>() {
        var index = -1
        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
            val v: PlaylistViewAllLayoutBinding = DataBindingUtil.inflate(
                LayoutInflater.from(parent.context),
                R.layout.playlist_view_all_layout,
                parent,
                false
            )
            return MyViewHolder(v)
        }

        @SuppressLint("SetTextI18n")
        override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
            val measureRatio = BWSApplication.measureRatio(activity, 0f, 1f, 1f, 0.44f, 0f)
            holder.binding.ivRestaurantImage.layoutParams.height =
                (measureRatio.height * measureRatio.ratio).toInt()
            holder.binding.ivRestaurantImage.layoutParams.width =
                (measureRatio.widthImg * measureRatio.ratio).toInt()
            holder.binding.ivRestaurantImage.scaleType = ImageView.ScaleType.FIT_XY
            holder.binding.tvAddToPlaylist.layoutParams.height =
                (measureRatio.height * measureRatio.ratio).toInt()
            holder.binding.tvAddToPlaylist.layoutParams.width =
                (measureRatio.widthImg * measureRatio.ratio).toInt()
            holder.binding.tvPlaylistName.text = listModelList[position].playlistName
            val measureRatio1 = BWSApplication.measureRatio(activity, 0f, 1f, 1f, 0.44f, 0f)
            holder.binding.rlMainLayout.layoutParams.height =
                (measureRatio1.height * measureRatio1.ratio).toInt()
            holder.binding.rlMainLayout.layoutParams.width =
                (measureRatio1.widthImg * measureRatio1.ratio).toInt()
            Glide.with(requireActivity())
                .load(listModelList[position].playlistImage)
                .thumbnail(0.05f)
                .apply(RequestOptions.bitmapTransform(RoundedCorners(42)))
                .priority(Priority.HIGH)
                .diskCacheStrategy(DiskCacheStrategy.ALL)
                .skipMemoryCache(false)
                .into(holder.binding.ivRestaurantImage)

            //            if (IsLock.equalsIgnoreCase("1")) {
            //                holder.binding.ivLock.setVisibility(View.VISIBLE);
            //            } else if (IsLock.equalsIgnoreCase("2")) {
            //                holder.binding.ivLock.setVisibility(View.VISIBLE);
            //            } else if (IsLock.equalsIgnoreCase("0") || IsLock.equalsIgnoreCase("")) {
            holder.binding.ivLock.visibility = View.GONE
            //            }
            if (index == position) {
                holder.binding.tvAddToPlaylist.visibility = View.VISIBLE
            } else holder.binding.tvAddToPlaylist.visibility = View.GONE
            holder.binding.tvAddToPlaylist.text = "Add To Playlist"
            holder.binding.rlMainLayout.setOnLongClickListener {
                holder.binding.tvAddToPlaylist.visibility = View.VISIBLE
                index = position
                notifyDataSetChanged()
                true
            }
            holder.binding.tvAddToPlaylist.setOnClickListener {
                holder.binding.ivLock.visibility = View.GONE
                val i = Intent(activity, AddPlaylistActivity::class.java)
                i.putExtra("AudioId", "")
                i.putExtra("ScreenView", "Playlist View All Screen")
                i.putExtra("PlaylistID", listModelList[position].playlistID)
                i.putExtra("PlaylistName", listModelList[position].playlistName)
                i.putExtra("PlaylistImage", listModelList[position].playlistImage)
                i.putExtra("PlaylistType", listModelList[position].created)
                i.putExtra("Liked", "0")
                startActivity(i)
            }
            holder.binding.rlMainLayout.setOnClickListener {
                //                if (IsLock.equalsIgnoreCase("1")) {
                //                    holder.binding.ivLock.setVisibility(View.VISIBLE);
                //                    Intent i = new Intent(getActivity(),
                // MembershipChangeActivity.class);
                //                    i.putExtra("ComeFrom", "Plan");
                //                    startActivity(i);
                //                } else if (IsLock.equalsIgnoreCase("2")) {
                //                    holder.binding.ivLock.setVisibility(View.VISIBLE);
                //                    BWSApplication.showToast(getString(R.string.reactive_plan),
                // getActivity());
                //                } else if (IsLock.equalsIgnoreCase("0") ||
                // IsLock.equalsIgnoreCase("")) {
                holder.binding.ivLock.visibility = View.GONE
                if (myDownloads.equals("1", ignoreCase = true)) {
                    //                            getMedia(listModelList.get(position).getPlaylistID());
                    val i = Intent(activity, DownloadPlaylistActivity::class.java)
                    i.putExtra("New", "0")
                    i.putExtra("PlaylistID", listModelList[position].playlistID)
                    i.putExtra("PlaylistName", listModelList[position].playlistName)
                    i.putExtra("PlaylistImage", listModelList[position].playlistImage)
                    i.putExtra("PlaylistImageDetails", "")
                    i.putExtra("TotalAudio", listModelList[position].totalAudio)
                    i.putExtra("Totalhour", listModelList[position].totalhour)
                    i.putExtra("Totalminute", listModelList[position].totalminute)
                    i.putExtra("MyDownloads", "1")
                    requireActivity().startActivity(i)
                } else {
                    GetPlaylistLibraryID = getLibraryId
                    val i = Intent(activity, MyPlaylistListingActivity::class.java)
                    i.putExtra("New", "0")
                    i.putExtra("PlaylistID", listModelList[position].playlistID)
                    i.putExtra("PlaylistName", listModelList[position].playlistName)
                    i.putExtra("PlaylistImage", listModelList[position].playlistImage)
                    i.putExtra("MyDownloads", myDownloads)
                    i.putExtra("ScreenView", screenView)
                    i.putExtra("PlaylistType", listModelList[position].created)
                    i.flags = Intent.FLAG_ACTIVITY_NO_ANIMATION
                    startActivity(i)
                }
            }
        }

        override fun getItemCount(): Int {
            return listModelList.size
        }

        inner class MyViewHolder(var binding: PlaylistViewAllLayoutBinding) :
            RecyclerView.ViewHolder(
                binding.root
            )
    }

    companion object {
        var GetPlaylistLibraryID: String? = ""
    }
}